Fix the epel repo install for rhel9

pwalczysko opened the pull request:
This PR is fixing an error happening on RHEL 9 when the ``ome.basedeps`` role is used.


```
TASK [ome.basedeps : system packages | install epel repo] ***********************************************************************
fatal: [134.36.4.7]: FAILED! => {"changed": false, "failures": ["No package epel-release available."], "msg": "Failed to install some of the specified packages", "rc": 1, "results": []}
```

We (@khaledk2 and @pwalczysko ) opted for the solution from https://github.com/ome/ansible-role-cli-utils/blob/master/tasks/main.yml == using the key and installing epel from fedora.

This solution works on RHEL 9 - tested via playbook.

pwalczysko committed:
Fix the epel repo install for rhel9

pwalczysko commented:
cc @jburel @sbesson

sbesson reviewed (commented):
Barring the double space issue which is failing the lint step, the proposed solution matches what was discussed in https://github.com/ome/ansible-role-cli-utils/pull/9.

Should the tasks be condition to the `ansible_os_family` being RedHat so that they don't get executed on Debian? Incidentally, this raises the question of whether the role should be tested on Ubuntu 22.04 (which is listed as a target distribution on the metadata file).

pwalczysko committed:
Add conditionals for install of epel via key for RedHat only

pwalczysko committed:
Fix the double-space cf Seb's comment

pwalczysko committed:
Fix "too many blank lines" error

pwalczysko commented:
> Should the tasks be condition to the `ansible_os_family` being RedHat so that they don't get executed on Debian?

Fixed in https://github.com/ome/ansible-role-basedeps/pull/13/commits/89030b444f68a80cfaece344ac8d764074647b1d

> the role should be tested on Ubuntu 22.04 (which is listed as a target distribution on the metadata file).

afai understand the output, the role is tested on Ubuntu 22.04

```
PLAY [Converge] ****************************************************************
  
  TASK [Gathering Facts] *********************************************************
  ok: [rockylinux-9]
  ok: [ubuntu-2204]
  
  TASK [ome.basedeps : Import a key for epel] ************************************
  skipping: [ubuntu-2204]
  changed: [rockylinux-9]
  
  TASK [ome.basedeps : epel | setup dnf repository] ******************************
  skipping: [ubuntu-2204]
  changed: [rockylinux-9]
  
  TASK [ome.basedeps : system packages | basic system utils (rockylinux)] ********
  skipping: [ubuntu-2204]
  changed: [rockylinux-9]
  
  TASK [ome.basedeps : system packages | basic system utils (ubuntu)] ************
  skipping: [rockylinux-9]
  changed: [ubuntu-2204]
  
  PLAY RECAP *********************************************************************
  rockylinux-9               : ok=4    changed=3    unreachable=0    failed=0    skipped=1    rescued=0    ignored=0
  ubuntu-2204                : ok=2    changed=1    unreachable=0    failed=0    skipped=3    rescued=0    ignored=0
```

pwalczysko commented:
All the errors from GA are fixed too, and the test is green see https://github.com/ome/ansible-role-basedeps/actions/runs/6880360622

I would suggest that we can merge and release if no objections ? cc @sbesson @jburel

khaledk2 commented:
I think It would be good if you upgrade the Python version from `3.8 `to `3.9` inside `molecule.yml`

pwalczysko committed:
Bump python to 3.9 cf Khaled's comment

pwalczysko commented:
> I think It would be good if you upgrade the Python version from `3.8 `to `3.9` inside `molecule.yml`

fixed in https://github.com/ome/ansible-role-basedeps/pull/13/commits/31dc4e69fd4cfd90ec3182e6d25a23eac5f9646b

pwalczysko commented:
Test is green again - should we then merge&release ? cc @jburel

khaledk2 commented:
You may use version 3 instead of version 2 to eliminate the remaining warning message
https://github.com/pwalczysko/ansible-role-basedeps/blob/31dc4e69fd4cfd90ec3182e6d25a23eac5f9646b/.github/workflows/molecule.yml#L17C1-L18C1

pwalczysko committed:
Bump actions/checkout to v3 cf Khaled's comment

pwalczysko committed:
Test runs on Ubuntu 22.04 to match other repos

pwalczysko commented:
> You may use version 3 instead of version 2 to eliminate the remaining warning message https://github.com/pwalczysko/ansible-role-basedeps/blob/31dc4e69fd4cfd90ec3182e6d25a23eac5f9646b/.github/workflows/molecule.yml#L17C1-L18C1

Thanks @khaledk2 , fixed in https://github.com/ome/ansible-role-basedeps/pull/13/commits/02c82ca5e0e7922436c557afee2b634592640291 - I also changed the OS on which the test is running to match your pattern from other role repos to 22.04 in https://github.com/ome/ansible-role-basedeps/pull/13/commits/da66db26645157e61f02fd5fbe89c08ecf15d859

pwalczysko commented:
@jburel do we want to merge and release to help with https://github.com/ome/omero-web-docker/pull/75 ?
